inference-engine/src/vpu/graph_transformer/src/stages/strided_slice.cpp

```diff
@@ -35,36 +52,95 @@ class StridedSliceStage final : public StageNode {
 
         std::vector<EnumSet<DataType>> expectedInputs3Types =
-            { {DataType::FP16}, {DataType::S32}, {DataType::S32} };
+            { {DataType::FP16, DataType::S32}, {DataType::S32}, {DataType::S32} };
         std::vector<EnumSet<DataType>> expectedInputs4Types =
-            { {DataType::FP16}, {DataType::S32}, {DataType::S32}, {DataType::S32} };
+            { {DataType::FP16, DataType::S32}, {DataType::S32}, {DataType::S32}, {DataType::S32} };
+
+        const auto& input0DataType = input(0)->desc().type();
 
         assertInputsOutputsTypes(
             this,
             numInputs() == 3 ? expectedInputs3Types : expectedInputs4Types,
-            {{DataType::FP16}});
+            {{input0DataType}});
     }
 
-    void serializeParamsImpl(BlobSerializer&) const override {
-        VPU_THROW_EXCEPTION << "Must never be called";
+    void serializeParamsImpl(BlobSerializer& serializer) const override {
+        std::string beginMask = origLayer()->GetParamAsString("begin_mask", "");
+        std::string endMask = origLayer()->GetParamAsString("end_mask", "");
+        serializer.append(maskStrToInt(beginMask));
+        serializer.append(maskStrToInt(endMask));
     }
 
-    void serializeDataImpl(BlobSerializer&) const override {
-        VPU_THROW_EXCEPTION << "Must never be called";
+    void serializeDataImpl(BlobSerializer& serializer) const override {
+        input(0)->serializeBuffer(serializer);
+        input(1)->serializeBuffer(serializer);
+        input(2)->serializeBuffer(serializer);
+        input(3)->serializeBuffer(serializer);
+        output(0)->serializeBuffer(serializer);
     }
 };
 
 }  // namespace
 
 void FrontEnd::parseStridedSlice(const Model& model, const ie::CNNLayerPtr& layer, const DataVector& inputs, const DataVector& outputs) const {
-    IE_ASSERT(inputs.size() == 3 || inputs.size() == 4);
-    IE_ASSERT(outputs.size() == 1);
+    VPU_THROW_UNLESS(inputs.size() == 3 || inputs.size() == 4,
+        "Parsing layer {} with type {} failed: number of input should be 3 or 4, but {} were provided",
+        layer->name, layer->type, inputs.size());
+    VPU_THROW_UNLESS(outputs.size() == 1,
+        "Parsing layer {} with type {} failed: number of outputs should be 1, but {} were provided",
+        layer->name, layer->type, outputs.size());
+
+    const auto& begin = inputs[1];
+    const auto& end = inputs[2];
+
+    VPU_THROW_UNLESS(begin->usage() == DataUsage::Const,
+                     "Checking {} with type {} failed: only {} type for begin is supported, but {} was provided",
+                     layer->name, layer->type, DataUsage::Const, begin->usage());
+    VPU_THROW_UNLESS(end->usage() == DataUsage::Const,
+                     "Checking {} with type {} failed: only {} type for end is supported, but {} was provided",
+                     layer->name, layer->type, DataUsage::Const, end->usage());
+    VPU_THROW_UNLESS(inputs.size() == 3 || inputs[3]->usage() == DataUsage::Const,
```

Comment:
In `initialCheckImpl` you consider that strides can be non-const

Reply:
Removed this check.